provider/mux_server.go

```diff
@@ -2,17 +2,18 @@ package provider
 
 import (
 	"context"
+	"github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
 	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
 	"github.com/hashicorp/terraform-plugin-mux/tf5muxserver"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-func MuxProviderServerCreator(sdkProvider *schema.Provider) (func() tfprotov5.ProviderServer, error) {
+func MuxProviderServerCreator(sdkProvider *schema.Provider, frameworkProvider provider.Provider) (func() tfprotov5.ProviderServer, error) {
```

Comment:
I think we might need to change this to _not_ take in the SDKv2 & framework providers as this repository is used by `https://github.com/RedisLabs/pulumi-rediscloud` and that'll need to construct the provider itself. My thinking is to make the change easy to whomever is maintaining that repository

Reply:
That might be tricky, a lot of the tests rely on a reference to the specific provider.
Although looking at it, it seems to just be used to get an apiClient...
I'll look into it after lunch 😄

Reply:
Perhaps the `MuxProviderServerCreator` method can return `(func() tfprotov5.ProviderServer, *apiClient, error)`